### PR TITLE
K8SPSMDB-316: Fix config service removing

### PR DESCRIPTION
**Pull request by blez**
[K8SPSMDB-316](https://jira.percona.com/browse/K8SPSMDB-316)

### CODE REVIEW

**Review by delgod — DISMISSED** [review]

File: pkg/controller/perconaservermongodb/psmdb_controller.go [hunk 1]
```diff
@@ -522,14 +522,24 @@ func (r *ReconcilePerconaServerMongoDB) deleteCfgIfNeeded(cr *api.PerconaServerM
...
 		return errors.Wrapf(err, "failed to delete sfs: %s", sfs.Name)
 	}
 
+	svc := corev1.Service{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: cr.Name + "-" + api.ConfigReplSetName, Namespace: cr.Namespace}, &svc)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrap(err, "failed to get config service")
+	}
+	err = r.client.Delete(context.TODO(), &svc)
+	if err != nil {
```
Comment (qJkee) [inline pkg/controller/perconaservermongodb/psmdb_controller.go:539]: should we also check for k8serrors.IsNotFound?
I think if object is not exists delete will also return this error

Comment (blez) [reply pkg/controller/perconaservermongodb/psmdb_controller.go:544]: https://github.com/percona/percona-server-mongodb-operator/pull/465/commits/06699e728959d2b0fc1122951e27090b6bc61286

**Review by tplavcic — APPROVED** [review]

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| upgrade | passed |
| storage | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| demand-backup | passed |
| monitoring-2-0 | failed |
| security-context | passed |
| self-healing | passed |
| arbiter | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-465-d4d7fb4d

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| storage | passed |
| security-context | passed |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-465-9fe8dd17

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| arbiter | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| one-pod | failed |
| storage | failed |
| upgrade | failed |
| self-healing | failed |
| limits | passed |
| monitoring | failed |
| upgrade-consistency | passed |
| scaling | failed |
| demand-backup | failed |
| security-context | failed |
| monitoring-2-0 | failed |
| operator-self-healing | failed |
| scheduled-backup | failed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | failed |
| demand-backup-sharded | failed |
| smart-update | failed |
| version-service | failed |
| users | failed |

**Pull request merged**